tests/src/core/testqgslayouttable.cpp

```diff
@@ -1052,6 +1054,34 @@ void TestQgsLayoutTable::verticalGrid()
   delete multiLineLayer;
 }
 
+void TestQgsLayoutTable::testDataDefinedTextFormatForCell()
+{
+  QgsLayout l( QgsProject::instance() );
+  l.initializeDefaults();
+  QgsLayoutItemAttributeTable *table = new QgsLayoutItemAttributeTable( &l );
+  table->setVectorLayer( mVectorLayer );
+
+  l.addMultiFrame( table );
+  QgsLayoutFrame *frame = new QgsLayoutFrame( &l, table );
+  frame->attemptSetSceneRect( QRectF( 5, 5, 100, 30 ) );
+  frame->setFrameEnabled( true );
+  l.addLayoutItem( frame );
+  table->addFrame( frame );
+
+  QgsTextFormat textFormat = QgsTextFormat::fromQFont( QgsFontUtils::getStandardTestFont( QStringLiteral( "Bold" ) ) );
+  table->setHeaderTextFormat( textFormat );
+  table->setContentTextFormat( textFormat );
+  table->refresh();
+
+  QCOMPARE( table->totalHeight(), 150.0 );
```

Comment:
This is likely to be fragile cross platform -- could we get a render test instead?

Reply:
I was trying to avoid a render test altogether. You think there's no way around it?

Reply:
I think it's preferable in this case -- it would also cover the actual rendering code then